docs/grammar/BallerinaParser.g4

```diff
@@ -622,3 +623,13 @@ reservedWord
     :   FOREACH
     |   TYPE_MAP
     ;
+
+query
+    :   FROM Identifier (AS Identifier)? (JOIN Identifier (AS Identifier)? ON expression)?
+        (WHERE expression)?
```

Comment:
`(WHERE expression)?` is too relaxed IMO. We should make it further strict to only allow the supported arithmetic and logical expressions.

Same applies for `(HAVING expression)?`

Reply:
@SupunS I agree. Basically the expression should be a conditional/Boolean expression. When I was having a discussion with @hasithaa, he mentioned that we can check for valid expressions in latter phases.